areader/reader.go

```diff
@@ -369,6 +369,11 @@ func (ar *Reader) readHeaderV3(tReader *tar.Reader,
 			return nil, errors.Wrap(err, "readHeaderV3")
 		}
 		if validPath {
+			// Artifact should be signed, but isn't, so do not process the update.
+			if ar.shouldBeSigned && !ar.IsSigned {
+				return nil,
+					errors.New("reader: expecting signed artifact, but no signature file found")
```

Comment:
2. This was found running the client tests. What it means is that we should not proceed with the update, if the client provides a key, and the artifact is not signed. Had not thought of this, good thing we have tests! (Enjoy this while u can)

Reply:
Haha, join the dark side! But yes, good catch indeed!